### PR TITLE
Move from dartfmt to dart format

### LINKED ISSUE #124: Update to call new `dart` tool
Several CLI commands are being deprecated from the Dart SDK (e.g. `dartfmt`, `dartanalyze`) in favor of a single `dart` comment; see: https://github.com/dart-lang/sdk/issues/46100

It would be great to update this plugin to use `dart` instead.

### PR DESCRIPTION
**Pull request by natebosch**
Closes #124

### CODE REVIEW

**Comment by natebosch** [comment]
This currently has terrible UX when the code has errors due to https://github.com/dart-lang/sdk/issues/46379

**Comment by fishythefish** [comment]
Is this unblocked since https://github.com/dart-lang/dart_style/issues/1035 is closed?

**Comment by natebosch** [comment]
Now this is blocked by https://github.com/dart-lang/sdk/issues/46636

After this is unblocked we'll need to figure out the rollout plan - given the bugs on existing stable SDKs I don't think we can roll this out universally. We might need to implement some kind of SDK version check. Or we could consider adding a setting which could be more annoying for users, but simpler to implement.

File: autoload/dart.vim [hunk 1]
```diff
@@ -60,14 +60,30 @@ function! dart#fmt(...) abort
...
 
 function! s:FindDartFmt() abort
-  if executable('dartfmt') | return 'dartfmt' | endif
+  if executable('dart')
+    let l:version_text = system('dart --version')
+    let l:match = matchlist(l:version_text,
+        \ '\vDart SDK version: (\d+)\.(\d+)\.\d+.*')
+    if empty(l:match)
+      call s:error('Unable to determine dart version')
+      return []
```
Comment (sigmundch) [inline autoload/dart.vim:69]: Rather than return with an error, would it be worth using the fallback in this case?

**Review by sigmundch — APPROVED** [review]
LGTM

File: autoload/dart.vim (hunk 1 above)
Comment (fishythefish) [reply autoload/dart.vim:69]: I think the error might be better. IIRC, it's difficult/tedious to distinguish between stdout and stderr in vim, so diagnostics emitted by `dart format`/`dartfmt` get inserted at the top of the file. In particular, if we fail to detect the Dart version and fall back to `dartfmt` when `dart format` is expected, the UX will be worse than not formatting at all.

Comment (sigmundch) [reply autoload/dart.vim:69]: got it - error sgtm.

Comment (natebosch) [reply autoload/dart.vim:69]: I don't expect any existing SDK releases to hit this. If the version string format changes in the future SDK releases, the fallback won't be the right behavior.

**Pull request merged**